tests/core/contracts/test_offchain_lookup.py

```diff
@@ -65,6 +66,53 @@ def test_offchain_lookup_functionality(
     assert decode_abi(['string'], response)[0] == 'web3py'
 
 
+def test_eth_call_offchain_lookup_raises_when_ccip_read_is_disabled(w3, offchain_lookup_contract):
+    # test ContractFunction call
+    with pytest.raises(_OffchainLookup):
+        offchain_lookup_contract.functions.testOffchainLookup(
+            OFFCHAIN_LOOKUP_TEST_DATA
+        ).call(ccip_read_enabled=False)
+
+    # test ContractCaller call
+    with pytest.raises(_OffchainLookup):
+        offchain_lookup_contract.caller(ccip_read_enabled=False).testOffchainLookup(
+            OFFCHAIN_LOOKUP_TEST_DATA
+        )
+
+    # test global flag on the provider
+    w3.provider.ccip_read_calls_enabled = False
+
+    with pytest.raises(_OffchainLookup):
+        offchain_lookup_contract.caller.testOffchainLookup(OFFCHAIN_LOOKUP_TEST_DATA)
+
+    with pytest.raises(_OffchainLookup):
+        offchain_lookup_contract.caller(ccip_read_enabled=False).testOffchainLookup(
```

Comment:
the `ccip_read_enabled=False` shouldn't be there, because you're testing the global flag, right?

Reply:
In this case I'm testing that `None` (default) and `False` values for the `call` flag should yield the same result when the global flag is set to `False`.

Reply:
dig it